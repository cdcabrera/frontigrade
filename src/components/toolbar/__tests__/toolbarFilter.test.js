import React from 'react';
import { mount } from 'enzyme';
import ToolbarFilter from '../toolbarFilter';

describe('ToolbarFilter Component', () => {
  it('should render a basic component', () => {
    const filterFields = [];

    const props = {
      filterFields,
      filterField: undefined
    };

    const component = mount(<ToolbarFilter {...props} />);
    expect(component).toMatchSnapshot('basic, no fields');

    component.setProps({
      filterFields: [
        {
          id: 'lorem',
          title: 'Lorem',
          placeholder: 'Filter by lorem type',
          filterType: 'text'
        }
      ],
      filterField: filterFields[0]
    });
    expect(component).toMatchSnapshot('basic, one field');
  });

  it('should render multiple filter fields', () => {
    const filterFields = [
      {
        id: 'lorem',
        title: 'Lorem',
        placeholder: 'Filter by lorem type',
        filterType: 'text'
      },
      {
        id: 'ipsum',
        title: 'Ipsum',
        placeholder: 'Filter by ipsum type',
        filterType: 'select',
        filterValues: [{ title: 'dolor', id: 'dolor' }, { title: 'sit', id: 'sit' }]
      }
    ];

    const props = {
      filterFields,
      filterField: filterFields[0]
    };

    const component = mount(<ToolbarFilter {...props} />);
    expect(component.render()).toMatchSnapshot('multiple filters, text');

    component.setProps({ filterField: filterFields[1] });
    expect(component.render()).toMatchSnapshot('multiple filters, select');
  });
});
