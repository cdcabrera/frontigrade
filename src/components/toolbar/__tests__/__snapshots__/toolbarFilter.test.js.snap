// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic, no fields 1`] = `
<ToolbarFilter
  filterField={null}
  filterFields={Array []}
  filterValue=""
  onFilterSelected={[Function]}
  onKeyPress={[Function]}
  onSelectFilter={[Function]}
  onUpdateValue={[Function]}
/>
`;

exports[`basic, one field 1`] = `
<ToolbarFilter
  filterField={null}
  filterFields={
    Array [
      Object {
        "filterType": "text",
        "id": "lorem",
        "placeholder": "Filter by lorem type",
        "title": "Lorem",
      },
    ]
  }
  filterValue=""
  onFilterSelected={[Function]}
  onKeyPress={[Function]}
  onSelectFilter={[Function]}
  onUpdateValue={[Function]}
>
  <getContext(Filter)>
    <Filter
      className=""
    >
      <div
        className="filter-pf form-group"
      >
        <div
          className="filter-pf-fields"
        >
          <div
            className="input-group"
          />
        </div>
      </div>
    </Filter>
  </getContext(Filter)>
</ToolbarFilter>
`;

exports[`multiple filters, select 1`] = `
<div
  class="filter-pf form-group"
>
  <div
    class="filter-pf-fields"
  >
    <div
      class="input-group"
    >
      <div
        class="input-group-btn"
      >
        <div
          class="dropdown btn-group"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-default"
            id="filterFieldTypeMenu"
            role="button"
            type="button"
          >
            Ipsum 
            <span
              class="caret"
            />
          </button>
          <ul
            aria-labelledby="filterFieldTypeMenu"
            class="dropdown-menu"
            role="menu"
          >
            <li
              class=""
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
              >
                Lorem
              </a>
            </li>
            <li
              class="selected"
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
              >
                Ipsum
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="filter-pf-select"
      >
        <div
          class="dropdown btn-group"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="filter-pf-select-dropdown dropdown-toggle btn btn-default"
            id="filterCategoryMenu"
            role="button"
            type="button"
          >
            Filter by ipsum type 
            <span
              class="caret"
            />
          </button>
          <ul
            aria-labelledby="filterCategoryMenu"
            class="dropdown-menu"
            role="menu"
          >
            <li
              class=""
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
                title="Filter by ipsum type"
              >
                Filter by ipsum type
              </a>
            </li>
            <li
              class=""
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
              >
                dolor
              </a>
            </li>
            <li
              class=""
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
              >
                sit
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`multiple filters, text 1`] = `
<div
  class="filter-pf form-group"
>
  <div
    class="filter-pf-fields"
  >
    <div
      class="input-group"
    >
      <div
        class="input-group-btn"
      >
        <div
          class="dropdown btn-group"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-default"
            id="filterFieldTypeMenu"
            role="button"
            type="button"
          >
            Lorem 
            <span
              class="caret"
            />
          </button>
          <ul
            aria-labelledby="filterFieldTypeMenu"
            class="dropdown-menu"
            role="menu"
          >
            <li
              class="selected"
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
              >
                Lorem
              </a>
            </li>
            <li
              class=""
              role="presentation"
            >
              <a
                href="#"
                role="menuitem"
                tabindex="-1"
              >
                Ipsum
              </a>
            </li>
          </ul>
        </div>
      </div>
      <input
        class="form-control"
        maxlength="128"
        placeholder="Filter by lorem type"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
